src/Personal/actionCreator.js

```
@@ -0,0 +1,144 @@
+  BACKUP_START: 'BACKUP_START',
+  BACKUP_SUCCESSFUL: 'BACKUP_SUCCESSFUL',
+  BACKUP_FAILED: 'BACKUP_FAILED',
+  SIGNUP_START: 'SIGNUP_START',
+  SIGNUP_SUCCESSFUL: 'SIGNUP_SUCCESSFUL',
+  SIGNUP_FAILED: 'SIGNUP_FAILED',
+  RESTORE_START: 'RESTORE_START',
+  RESTORE_SUCCESSFUL: 'RESTORE_SUCCESSFUL',
+  RESTORE_FAILED: 'RESTORE_FAILED',
+};
+
+export function backupStart() {
+  return {
+    type: actionType.BACKUP_START,
+  };
+}
+
+export function backupSuccessful(timestamp) {
+  return {
+    type: actionType.BACKUP_SUCCESSFUL,
+    payload: timestamp,
+  };
+}
+
+export function backupFailed(errorMsg) {
+  return {
+    type: actionType.BACKUP_FAILED,
+    payload: errorMsg,
+  };
+}
+
+export function signupStart() {
+  return {
+    type: actionType.SIGNUP_START,
+  };
+}
+
+export function signupSuccessful(userInfo) {
+  return {
+    type: actionType.signupSuccessful,
+    payload: userInfo,
+  };
+}
+
+export function signupFailed() {
+  return {
+    type: actionType.signupFailed,
+  };
+}
+
+export function restoreStart() {
+  return {
+    type: actionType.RESTORE_START,
+  };
+}
+
+export function restoreSuccessful(payload) {
+  return {
+    type: actionType.RESTORE_SUCCESSFUL,
+    payload,
+  };
+}
+
+export function restoreFailed(payload) {
+  return {
+    type: actionType.RESTORE_FAILED,
+    payload,
+  };
+}
+
+export const requestBackup = () => (dispatch, getState) => {
+  const { transactions } = getState().transactions;
+  if (!accessToken) {
+    dispatch(backupFailed('Unauthorized token'));
+    return;
+  }
+  dispatch(backupStart());
+  fetch(`${BACKEND_URL}/backup`, {
+    method: 'POST',
+    mode: 'cors',
+    headers: {
+      Authorization: `Bearer ${accessToken}`,
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({ transactions }),
+  }).then((response) => {
+    if (response.status === 200) {
+      setTimeout(() => dispatch(backupSuccessful(moment().unix())), 300);
+      throw new Error(200);
```

why do you need to throw an error here?